### ISSUE TITLE
setBackgroundDrawable only works when that day is clicked

### ISSUE THREAD

**Issue by oliviadodge** [prolificinteractive/material-calendarview#308]
Hi, I'm trying to do something similar to BasicActivityDecorated.java and decorate today's date. The only two differences are that I want to use a custom drawable for the decorator and I am using a DialogFragment. When I open the dialog, I don't see today's date decorated with my custom drawable until I click on it. Then I see the decorator. I'm wondering if anyone could help me with this problem. Thanks in advance!

**Comment by quentin41500** [comment]
Hi @oliviadodge, first of all, i don't think setBackgroundDrawable is the right method to use. You should use `widget.addDecorators(new MySelectorDecorator(this));` with `MySelectorDecorator` looking like this:
`public class MySelectorDecorator implements DayViewDecorator {

    private final Drawable drawable;

    public MySelectorDecorator(Activity context) {
        drawable = context.getResources().getDrawable(R.drawable.my_selector);
    }

    @Override
    public boolean shouldDecorate(CalendarDay day) {
        return true;
    }

    @Override
    public void decorate(DayViewFacade view) {
        view.setSelectionDrawable(drawable);
    }
}
`
Finally, you should set `widget.setSelectedDate(CalendarDay.today());` to select today's date.

Let me know if that works for you.

**Comment by quentin41500** [comment]
my_selector looks like this

```
<?xml version="1.0" encoding="utf-8"?>
<selector xmlns:android="http://schemas.android.com/apk/res/android"
    android:exitFadeDuration="@android:integer/config_shortAnimTime">

    <item android:state_checked="true"
        android:drawable="@drawable/ic_my_selector" />

    <item android:state_pressed="true"
        android:drawable="@drawable/ic_my_selector" />

    <item android:drawable="@android:color/transparent" />

</selector>
```

**Comment by oliviadodge** [comment]
Hi @quentin41500, thanks so much for your quick response! I actually found the source of the problem in the selector file. I forgot that if you only have the state_checked="true" and the state_pressed="true", the icon will only show up if you press the DayView. I set the same icon on state_checked="false" and state_pressed="false" and it works for me. Or I probably don't even need the selector file since I want the  icon to show up no matter what.
